### PR TITLE
Tenant improvements: IdP identifiers, person fields

### PR DESCRIPTION
**Pull request by matej21**
This change is [Reviewable](https://reviewable.io/reviews/contember/engine/295)

### CODE REVIEW

**Review by JanTvrdik — CHANGES_REQUESTED** [review]

**Review by JanTvrdik — APPROVED** [review]
> Reviewed 46 of 46 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/engine/295)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @matej21)

**Review by JanTvrdik — CHANGES_REQUESTED** [review]
x

File: packages/engine-tenant-api/src/model/service/IDPSignInManager.ts [hunk 1]
```diff
@@ -79,6 +67,40 @@ class IDPSignInManager {
...
 	}
+
+	private async resolvePerson(db: DatabaseContext, claim: IDPClaim, provider: IdentityProviderRow): Promise<PersonRow | null> {
+		const personByIdPQuery = new PersonByIdPQuery(provider.id, claim.externalIdentifier)
+		const personByIdp = await db.queryHandler.fetch(personByIdPQuery)
+		if (personByIdp) {
+			return personByIdp
+		}
+
+		const saveIdpIdentifier = async (personId: string) =>
```
[the remark this answers is not in the document]
Comment (matej21) [reply packages/engine-tenant-api/src/model/service/IDPSignInManager.ts:78]: This is much more compact, the private method would require 4 arguments.

File: packages/engine-tenant-api/src/migrations/2022-07-07-141000-person-display-name.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE person
+	ADD name TEXT;
+UPDATE person
+SET name = email;
```
[the remark this answers is not in the document]
Comment (matej21) [reply packages/engine-tenant-api/src/migrations/2022-07-07-141000-person-display-name.sql:4]: part before `@` is enough, i think

File: packages/engine-tenant-api/src/model/service/OtpManager.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ export class OtpManager {
 	) {}
 
 	async prepareOtp(dbContext: DatabaseContext, person: PersonRow, label: string): Promise<OtpData> {
-		const otp = await this.otpAuthenticator.create(person.email ?? 'unnamed', label)
+		const otp = await this.otpAuthenticator.create(person.email ?? person.name ?? 'unnamed', label)
```
[the remark this answers is not in the document]
Comment (matej21) [reply packages/engine-tenant-api/src/model/service/OtpManager.ts:12]: email is imho better identifier here

**Review by JanTvrdik — COMMENTED** [review]
> Reviewed 2 of 42 files at r2, 2 of 11 files at r3, 16 of 22 files at r4, 14 of 16 files at r5, 8 of 8 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/engine/295)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @matej21)

File: packages/engine-tenant-api/src/model/service/IDPSignInManager.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JanTvrdik) [reply packages/engine-tenant-api/src/model/service/IDPSignInManager.ts:78]: does not matter, you can say that about most private methods. they are used for code structure, not to save bytes

**Review by JanTvrdik — APPROVED** [review]
> Reviewed 2 of 2 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/engine/295)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @matej21)

**Pull request merged**